core/collection_mode.ml

```diff
@@ -1,28 +1,95 @@
 open! Core
 
+module Primary = struct
+  type t =
+    | Intel_processor_trace
+    | Stacktrace_sampling
+end
+
+module Event = struct
+  module Name = struct
+    type t =
+      | Branch_misses
+      | Cache_misses
+    [@@deriving compare, hash, sexp]
+
+    let to_string = function
+      | Branch_misses -> "branch-misses"
+      | Cache_misses -> "cache-misses"
+    ;;
+  end
+
+  module When_to_sample = struct
+    type t =
+      | Frequency of int
+      | Period of int
+    [@@deriving of_sexp]
+  end
+
+  module Precision = struct
+    type t =
+      | Arbitrary_skid
+      | Constant_skid
+      | Request_zero_skid
+      | Zero_skid
+      | Maximum_possible
+    [@@deriving of_sexp]
+  end
+
+  type t =
+    { when_to_sample : When_to_sample.t
+    ; name : Name.t
+    ; precision : Precision.t
+    }
+  [@@deriving of_sexp]
+
+  let of_string = function
+    | "branch-misses" ->
+      { when_to_sample = Period 50; name = Branch_misses; precision = Maximum_possible }
+    | "cache-misses" ->
+      { when_to_sample = Period 50; name = Cache_misses; precision = Maximum_possible }
+    | str -> t_of_sexp (Sexp.of_string str)
```

Comment:
@Xyene a couple things to think about here. One, what do you think are good defaults for precision and when_to_sample? Two, do you think there should be an alternative to the sexp in order to control when_to_sample / precision?